### PR TITLE
Fix wrong labels and annotations in kubernetes discovery native client

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java [hunk 1]
```diff
@@ -112,27 +113,7 @@ public List<ServiceInstance> getInstances(String serviceId) {
 	}
 
 	private Stream<ServiceInstance> getServiceInstanceDetails(V1Service service, String serviceId) {
-		Map<String, String> svcMetadata = new HashMap<>();
```
Comment (wind57) [inline spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java:115]: this PR fixes an issue, imo. if you look at this piece of code: 

```
service.getMetadata().getAnnotations().entrySet().stream()
							.filter(e -> e.getKey().startsWith(annotationPrefix))
							.forEach(e -> svcMetadata.put(e.getKey(), e.getValue()));
```

it says that we are going to take only annotations that start with `annotationPrefix`. In the fabric8 implementation we do it differently: we prefix existing values with it. As a matter of fact we do this in configmaps and secrets also. 

As such, I removed this code and fixed the bug in the utils class, where the code is the same as in the fabric8 implementation. Also added tests.

File: spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesDiscoveryClientUtils.java [hunk 2]
```diff
@@ -61,4 +67,38 @@ static boolean matchesServiceLabels(V1Service service, KubernetesDiscoveryProper
...
+			serviceMetadata.putAll(labelMetadata);
+		}
+		if (metadataProps.addAnnotations()) {
+			Map<String, String> annotationMetadata = keysWithPrefix(service.getMetadata().getAnnotations(),
+					metadataProps.annotationsPrefix());
+			LOG.debug(() -> "Adding annotations metadata: " + annotationMetadata + " for serviceId: " + serviceId);
+			serviceMetadata.putAll(annotationMetadata);
+		}
+
+		serviceMetadata.put(NAMESPACE_METADATA_KEY,
```
Comment (wind57) [inline spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesDiscoveryClientUtils.java:96]: metadata now includes two more fields, just like we do in the fabric8 implementation.

**Comment by wind57** [comment]
this PR. has a dependency on [this one](https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1276), because I want to add an integration test that would assert service metadata, and in that PR I created already an integration test, so I'll wait for it to be reviewed.

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-discovery-it/src/test/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesClientDiscoveryPodMetadataIT.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+				}).retryWhen(retrySpec()).block();
+
+		Assertions.assertEquals(ourServiceInstances.size(), 1);
+
+		DefaultKubernetesServiceInstance serviceInstance = ourServiceInstances.get(0);
+		Assertions.assertNotNull(serviceInstance.getInstanceId());
+		Assertions.assertEquals(serviceInstance.getServiceId(), "spring-cloud-kubernetes-client-discovery-it");
+		Assertions.assertNotNull(serviceInstance.getHost());
+		Assertions.assertEquals(serviceInstance.getMetadata(),
+				Map.of("http", "8080", "k8s_namespace", "default", "type", "ClusterIP",
```
Comment (wind57) [inline spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-discovery-it/src/test/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesClientDiscoveryPodMetadataIT.java:121]: an integration test that asserts for proper labels and annotations with prefix

**Comment by wind57** [comment]
@ryanjbaxter this one can be looked at too. I started to fix some issues as I dive into the implementation and have fabric8 context in my head

**Review by ryanjbaxter — APPROVED** [review]

**Pull request merged**